test/ussd_registration.test.js

```diff
@@ -0,0 +1,578 @@
+                            "1. Monday and Wednesday",
+                            "2. Tuesday and Thursday"
+                        ].join('\n')
+                    })
+                    .run();
+            });
+            it("to state_receive_calls_time", function() {
+                return tester
+                    .setup.user.addr('082111')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '12345'   // state_auth_code - personnel code
+                        , '0803304899' // state_msisdn - mobile number
+                        , '1'  // state_msg_receiver - mother
+                        , '1'  // state_msg_pregnant - mother
+                        , '3'  // state_last_period_month - May 15
+                        , '12' // state_last_period_day - 12
+                        , '1'  // state_msg_language - english
+                        , '1'   // state_msg_call_or_text - voice calls
+                        , '2'   // state_receive_calls_days - tuesdays and thursdays
+                    )
+                    .check.interaction({
+                        state: 'state_receive_calls_time',
+                        reply: [
+                            "Thank you. At what time would they like to receive these calls?",
+                            "1. Between 9-11am",
+                            "2. Between 2-5pm"
+                        ].join('\n')
+                    })
+                    .run();
+            });
+            it("complete flow - mother pregnant, voice calls", function() {
+                return tester
+                    .setup.user.addr('082111')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '12345'   // state_auth_code - personnel code
+                        , '0803304899' // state_msisdn - mobile number
+                        , '1'  // state_msg_receiver - mother
+                        , '1'  // state_msg_pregnant - mother
+                        , '3'  // state_last_period_month - May 15
+                        , '12' // state_last_period_day - 12
+                        , '1'  // state_msg_language - english
+                        , '1'   // state_msg_call_or_text - voice calls
+                        , '2'   // state_receive_calls_days - tuesdays and thursdays
+                        , '2'   // state_receive_calls_time - between 2-5pm
+                    )
+                    .check.interaction({
+                        state: 'state_end_thank_you_calls',
+                        reply: "Thank you. The person will now start receiving calls on [day and day] between [time - time]."
+                    })
+                    .run();
+            });
+        });
+
+        // flow testing case 2 - mother has baby and opts for text sms's
+        describe("Flow testing - mother with baby, text sms's", function() {
```

Comment:
I think I may have half-answered one of your questions - let's discuss testing duplication face to face.